### PR TITLE
Custom layout

### PR DESCRIPTION
**Pull request by jvc56**

### CODE REVIEW

**Comment by jvc56** [comment]
On second look it does seem like we need to account for bricks in board_is_position_valid

File: src/ent/board_layout.c [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+board_layout_load_status_t
+board_layout_parse_split_start_coords(BoardLayout *bl,
+                                      const StringSplitter *starting_coords) {
+
+  if (string_splitter_get_number_of_items(starting_coords) != 2) {
+    return BOARD_LAYOUT_LOAD_STATUS_MALFORMED_START_COORDS;
+  }
+
+  for (int i = 0; i < 2; i++) {
+    int lane_start_value =
```
Comment (olaugh) [inline src/ent/board_layout.c:81]: const

File: src/ent/board_layout.c [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+uint8_t bonus_square_char_to_value(char bonus_square_char) {
+  return bonus_square_chars_to_values_map[(int)bonus_square_char];
+}
+
+board_layout_load_status_t
+board_layout_parse_split_file(BoardLayout *bl,
+                              const StringSplitter *file_lines) {
+  int number_of_rows = string_splitter_get_number_of_items(file_lines);
```
Comment (olaugh) [inline src/ent/board_layout.c:102]: const

File: src/ent/board_layout.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    return status;
+  }
+
+  for (int row = 0; row < BOARD_DIM; row++) {
+    const char *layout_row = string_splitter_get_item(file_lines, row + 1);
+    if (string_length(layout_row) != BOARD_DIM) {
+      return BOARD_LAYOUT_LOAD_STATUS_INVALID_NUMBER_OF_COLS;
+    }
+    for (int col = 0; col < BOARD_DIM; col++) {
+      char bonus_square_char = layout_row[col];
```
Comment (olaugh) [inline src/ent/board_layout.c:126]: const for all these

File: test/tsrc/test.c [hunk 4]
```diff
@@ -134,14 +139,31 @@ void run_test(const char *subtest) {
...
+    if (argc == 1) {
+      run_all();
+    } else {
+      for (int i = 1; i < argc; i++) {
+        run_test(argv[i]);
+      }
     }
+  } else if (BOARD_DIM == DEFAULT_SUPER_BOARD_DIM) {
+    if (argc > 1) {
+      log_warn("Ignoring test argumentss when testing default super board "
```
Comment (olaugh) [inline test/tsrc/test.c:157]: argumentss typo

**Review by olaugh — APPROVED** [review]

**Pull request merged**